tests/postgres_tests/test_array.py

```diff
@@ -414,6 +414,24 @@ def test_overlap_charfield_including_expression(self):
             [obj_1, obj_2],
         )
 
+    def test_overlap_values_array_field(self):
+        # issue 26056
+        post1 = CharArrayModel(field=["django"])
+        post2 = CharArrayModel(field=["thoughts"])
+        post3 = CharArrayModel(field=["tutorial"])
+        CharArrayModel.objects.bulk_create([post1, post2, post3])
+        qs = CharArrayModel.objects.filter(
+            field__overlap=CharArrayModel.objects.values_list("field", flat=True)
+        )
+        self.assertSequenceEqual(
+            qs.values_list("field", flat=True),
+            [
+                (["django"]),
+                (["thoughts"]),
+                (["tutorial"]),
+            ],
+        )
```

Comment:
I'd reuse existing objects and add `filter()` to ensure that it actually works, e.g.:
```suggestion
        values = NullableIntegerArrayModel.objects.filter(order__lt=3).values_list(
            "field"
        )
        self.assertCountEqual(
            NullableIntegerArrayModel.objects.filter(
                field__overlap=values,
            ),
            self.objs[:3],
        )
```